### PR TITLE
scripts: kconfig: Extend the assignment-failed warning

### PR DESCRIPTION
**Pull request by ulfalizer**
- Tailor the warning when the symbol has no prompt, explaining how
   promptless symbols get values. Add some anti-select propaganda too.

 - Reference the 'Setting configuration values' in the Board Porting
   Guide. It explains Kconfig.defconfig files.

Fixes: #8388

### CODE REVIEW

**Comment by ulfalizer** [comment]
CC @carlescufi

File: scripts/kconfig/kconfig.py [hunk 1]
```diff
@@ -97,6 +98,28 @@ def main():
     kconf.write_autoconf(args.autoconf)
 
 
+# Message printed when a promptless symbol is assigned (and doesn't get the
+# assigned value)
+PROMPTLESS_HINT = """
+This symbol has no prompt, meaning direct assignments have no effect on it. It
```
Comment (SebastianBoe) [inline scripts/kconfig/kconfig.py]: We don't use the term "direct assignment" in our documentation on application configuration.

Perhaps we can mention the term "configuration file" as that is mentioned.

So something like 

> This symbol has no prompt, meaning direct assignments by configuration files have no effect.

Comment (SebastianBoe) [reply scripts/kconfig/kconfig.py]: Alternatively we could update the docs ...

**Review by SebastianBoe — APPROVED** [review]

**Comment by ulfalizer** [comment]
@SebastianBoe 
Updated it to say "assignments in configuration files".

I increased the `textwrap` width to 100 too. Just noticed that the symbol reference link was being broken across two lines.

**Review by nashif — APPROVED** [review]

**Pull request merged**